qt/qt_common/map_widget.cpp

```diff
@@ -383,8 +383,21 @@ void MapWidget::initializeGL()
     else
     {
       LOG(LINFO, ("Contex is LibGL"));
-      // TODO: Separate apiOpenGL3 from apiOpenGLES3, and use that for the currend shader code.
-      m_apiOpenGLES3 = true;
+      GLint majorVersion = 0, minorVersion = 0;
+      funcs->glGetIntegerv(GL_MAJOR_VERSION, &majorVersion);
+      funcs->glGetIntegerv(GL_MINOR_VERSION, &minorVersion);
+
+      if (majorVersion < 3 || (majorVersion == 3 && minorVersion < 2))
```

Comment:
Don't you want to use QPair here?
```
qMakePair(funcs->glGetIntegerv(GL_MAJOR_VERSION, &majorVersion), funcs->glGetIntegerv(GL_MINOR_VERSION, &minorVersion)) < qMakePair(3, 2)
```
https://doc.qt.io/qt-5/qpair.html#operator-lt

Reply:
I took the alternative